### DISCUSSION TITLE
在MAC電腦上安裝Python3與JupyterLab

### DISCUSSION THREAD

**Discussion by chia-kai-chang** [1101-GS4719-C/Python-Community-GS4719-C#367] [Ideas]
全新的MAC電腦上面已經有Python2與Python3的軟體了，如果要確認一下Python3軟體的版本，可以在「終端機」輸入指令 `python3` ，進入Python3的環境，就會顯示Python3軟體的版本，目前我這邊的版本是3.8.9。

![image](https://user-images.githubusercontent.com/1768891/153559312-08a70af9-eacf-4b29-8493-07ceba4a00ac.png)

如果想確定目前的Python3的實際路徑，可以依序輸入以下程式：
```python
>>> import sys
>>> sys.executable
```

![image](https://user-images.githubusercontent.com/1768891/153559344-92182972-0581-420f-965e-b3070c205dc5.png)

如果想安裝JupyterLab，可以輸入以下指令：
```
pip3 install jupyterlab
```

預設會安裝到以下目錄：
```
/Users/{自己帳戶名稱}/Library/Python/3.8/bin
```

此時如果在終端機輸入 `jupyter lab` 試圖啟動jupyterlab，會發生錯誤訊息，訊息內容是找不到 `jupyter` 這個執行檔。這是因為系統環境變數中 `$PATH` 尚未包含 `/Users/{自己帳戶名稱}/Library/Python/3.8/bin` 這個路徑，所以要手動把這個路徑加到環境變數中。

可以先在終端機輸入以下指令，查看目前的PATH設定：
```
echo $PATH
```
對於MAC來說，預設的系統環境變數如下：
```
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

我們要編輯 `~/.zshrc` 檔案，把 Jupyter 的安裝目錄加到系統環境變加到系統環境變數中。

可以在終端機輸入以下指令，進入編輯界面：
```
nano ~/.zshrc
```

在編輯介面中，加入以下的一行路徑設定，接著按下 <ctrl + X>，儲存並離開這個文字編輯界面。

```
export PATH=$PATH:/Users/benjamin/Library/Python/3.8/bin
```

![image](https://user-images.githubusercontent.com/1768891/153580539-c4c3b4bb-15d3-405a-95b6-cecfc3f6297c.png)

重新開機就可以讓這個路徑設定進去，如果不想重新開機，可以執行以下程式，強制把這段路徑設定吃進系統環境變數。
```
source ~/.zshrc
```

**Comment by enoch-lightning** [comment]
我沒有mac電腦，這樣就沒辦法嘗試了。
